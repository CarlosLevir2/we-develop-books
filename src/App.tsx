import { SkeletonTheme } from 'react-loading-skeleton';
import Provider from './Provider';
import AppRoutes from './routes';
import { useUserPreferences } from './store/useUserPreferences';
import GlobalStyles from './styles/global';
import theme from './styles/theme';

function App() {
  const { themeName } = useUserPreferences();

  return (
    <Provider>
      <SkeletonTheme highlightColor={theme[themeName].colors.accent.soft}>
        <GlobalStyles />
        <AppRoutes />
      </SkeletonTheme>
    </Provider>
  );
}

export default App;
