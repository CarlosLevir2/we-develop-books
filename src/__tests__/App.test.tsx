import { render, screen } from '@testing-library/react';
import App from '../App';

test('renders login page', () => {
  render(<App />);
  const linkElement = screen.getByText(/Login/i);
  expect(linkElement).toBeInTheDocument();
});

test('match snapshot', () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});
