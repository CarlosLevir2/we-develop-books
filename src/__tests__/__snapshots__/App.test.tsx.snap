// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div>
  <div
    class="sc-jrQzAO iRfREX"
  >
    <div
      class="sc-kDTinF ctUWHg"
    >
      <input
        class="sc-iqseJM bnnAZf"
        placeholder="Email"
      />
      <input
        class="sc-iqseJM bnnAZf"
        placeholder="Password"
      />
      <button
        class="sc-crHmcD hlgrjc"
      >
        Login
      </button>
    </div>
  </div>
</div>
`;
